### PR TITLE
config: add configuration entry make TiDB version string configurable

### PR DESCRIPTION
**Pull request by reafans**
### What problem does this PR solve? 
It can improve the hard-coding of `serverVersion `property. 

### What is changed and how it works?
add the `serverVersion `in `config/config.go` and `config/config.toml.example`

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported variable/fields change

### CODE REVIEW

File: tidb-server/main.go [hunk 1]
```diff
@@ -79,6 +79,7 @@ const (
 	nmSocket           = "socket"
 	nmEnableBinlog     = "enable-binlog"
 	nmRunDDL           = "run-ddl"
+	nmServerVersion    = "server-version"
```
Comment (crazycs520) [inline tidb-server/main.go]: useless line, remove this.

File: tidb-server/main.go [hunk 2]
```diff
@@ -154,6 +155,7 @@ func main() {
 	registerMetrics()
 	configWarning := loadConfig()
 	overrideConfig()
+	mysql.ServerVersion = cfg.ServerVersion
```
Comment (crazycs520) [inline tidb-server/main.go]: move to loadConfig()? and add a check for` cfg.ServerVersion` is not "".

File: config/config.go [hunk 3]
```diff
@@ -19,6 +19,7 @@ import (
 	"crypto/x509"
 	"encoding/json"
 	"fmt"
+	"github.com/pingcap/parser/mysql"
```
Comment (crazycs520) [inline config/config.go]: move this below #30 line, and fmt code.

File: config/config.go [hunk 4]
```diff
@@ -445,6 +446,7 @@ var defaultConf = Config{
 		Capacity: 2048000,
 	},
 	LowerCaseTableNames: 2,
+	ServerVersion:       "5.7.25-TiDB",
```
Comment (crazycs520) [inline config/config.go]: the default value of this should be ""

File: go.sum [hunk 5]
```diff
@@ -1,8 +1,8 @@
 cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
 github.com/BurntSushi/toml v0.3.1 h1:WXkYYl6Yr3qBf1K79EBnL4mak0OimBfB0XUf9Vl28OQ=
 github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=
-github.com/StackExchange/wmi v0.0.0-20190523213315-cbe66965904d h1:G0m3OIz70MZUWq3EgK3CesDbo8upS2Vm9/P3FtgI+Jk=
-github.com/StackExchange/wmi v0.0.0-20190523213315-cbe66965904d/go.mod h1:3eOhrUMpNV+6aFIbp5/iudMxNCF27Vw2OZgy4xEx0Fg=
+github.com/StackExchange/wmi v0.0.0-20180725035823-b12b22c5341f h1:5ZfJxyXo8KyX8DgGXC5B7ILL8y51fci/qYz2B4j8iLY=
```
Comment (crazycs520) [inline go.sum]: why change this file?

File: tidb-server/main.go [hunk 6]
```diff
@@ -154,6 +154,7 @@ func main() {
 	registerMetrics()
 	configWarning := loadConfig()
 	overrideConfig()
+
```
Comment (crazycs520) [inline tidb-server/main.go]: remove this line.

File: config/config.go [hunk 7]
```diff
@@ -638,6 +640,10 @@ func (c *Config) Load(confFile string) error {
 	if c.TokenLimit == 0 {
 		c.TokenLimit = 1000
 	}
+	if c.ServerVersion == "" {
+		c.ServerVersion = mysql.ServerVersion
+	}
+	mysql.ServerVersion = c.ServerVersion
```
Comment (crazycs520) [inline config/config.go]: How about 
```go
if len(c.ServerVersion) > 0 {
		mysql.ServerVersion = c.ServerVersion
}
```

File: config/config_test.go [hunk 8]
```diff
@@ -15,6 +15,7 @@ package config
 
 import (
 	"encoding/json"
+	"github.com/pingcap/parser/mysql"
```
Comment (crazycs520) [inline config/config_test.go]: move this after line#24, and fmt code.

**Comment by crazycs520** [comment]
@reafans Please resolve conflict and address comment.

File: config/config_test.go [hunk 9]
```diff
@@ -364,5 +368,6 @@ func (s *testConfigSuite) TestTxnTotalSizeLimitValid(c *C) {
 	for _, tt := range tests {
 		conf.Performance.TxnTotalSizeLimit = tt.limit
 		c.Assert(conf.Valid() == nil, Equals, tt.valid)
+
```
Comment (crazycs520) [inline config/config_test.go]: remove this blank line.

File: config/config_test.go [hunk 10]
```diff
@@ -220,8 +224,8 @@ max-sql-length=1024
 
 	configFile = filepath.Join(filepath.Dir(localFile), "config.toml.example")
 	c.Assert(conf.Load(configFile), IsNil)
-
 	// Make sure the example config is the same as default config.
+	conf.ServerVersion = ""
```
Comment (crazycs520) [inline config/config_test.go]: can we remove this?

File: config/config.toml.example [hunk 11]
```diff
@@ -74,6 +74,9 @@ split-region-max-num = 1000
 # In order to support "drop primary key" operation , this flag must be true and the table does not have the pkIsHandle flag.
 alter-primary-key = false
 
+# serverVersion is used to show the version of tidb when log in.
```
Comment (bb7133) [inline config/config.toml.example]: Please add the comment to explain the version when `server-version = ""`(Original TiDB version)

**Comment by bb7133** [comment]
Notice that the result of `select @@version` is not affected, I'm not sure if this is good enough for this feature.

**Pull request merged**